File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    for i in arr:
        if len(answer) == 0:
            answer.append(i)
            continue
        else:
            item = answer.pop()
            if i == item:
                answer.append(item)
            else:
                answer.append(item)
                answer.append(i)
        
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();

        for (int now : arr) {
            if (!stack.isEmpty()) {
                if (stack.peek() != now)
                    stack.push(now);
            } else {
                stack.push(now);
            }
        }

        int size = stack.size();

        int[] answer = new int[size];

        for (int i = size - 1; i >= 0; i--) {
            answer[i] = stack.pop();
        }

        return answer;
    }
}